hyperon_das_atomdb/database.py

```diff
@@ -474,7 +486,7 @@ def get_all_nodes(self, node_type: str, names: bool = False) -> list[str]:
         """
 
     @abstractmethod
-    def get_all_links(self, link_type: str, **kwargs) -> list[str] | tuple[int, list[str]]:
+    def get_all_links(self, link_type: str, **kwargs) -> tuple[int | None, list[str]]:
```

Comment:
I might be wrong here:
```suggestion
    def get_all_links(self, link_type: str, **kwargs) -> MatchedLinksResultT:
```

Reply:
`MatchedLinksResultT` is more than just `tuple[int | None, list[str]]` - it accepts/allows a different structure as the second element (`MatchedTargetsListT`).